lib/board.rb

```diff
@@ -25,36 +23,35 @@ def valid_coordinate?(coordinate)
   end
 
   def valid_placement?(ship, coordinates)
+    return false unless ship.length == coordinates.count && valid_coordinate?(coordinates.first)
     acceptable_placements = []
-    coordinate = coordinates.first
-    acceptable_combo = []
-    if ship.length == coordinates.count && valid_coordinate?(coordinate)
-      acceptable_combo << coordinate #CHECKING FIRST COORDINATE
-      #GOING RIGHT
-      (ship.length-1).times do 
-        if valid_coordinate?(coordinate.next) == true 
-        acceptable_combo << coordinate.next 
-        coordinate = coordinate.next
-        end
-      end
-      acceptable_placements << acceptable_combo
-      #GOING DOWN
-      coordinate = coordinates.first
-      acceptable_combo = [coordinate]
-      (ship.length-1).times do 
-        coordinate = coordinate.delete("^A-Z").next + coordinate.delete("^0-9")
-        if valid_coordinate?(coordinate) == true 
-          acceptable_combo << coordinate 
-        end
-      end
-      acceptable_placements << acceptable_combo
-      acceptable_placements = acceptable_placements.find_all {|combo| combo.count == ship.length} 
-      acceptable_placements.include?(coordinates)
-      # require 'pry'; binding.pry
-    else
-      false
+    coord_letter = coordinates.first.delete("^A-Z")
```

Comment:
Refactoring 25 lines of code vs 11 is a landmark achievement. Using what the interaction pattern has to offer is always a smart strategy before re-inventing the wheel.  Glad you saw that the return value was needed to fix the bug in the method.